src/transformers/hf_argparser.py

```diff
@@ -175,8 +175,10 @@ def _parse_dataclass_field(parser: ArgumentParser, field: dataclasses.Field):
         # A variable to store kwargs for a boolean field, if needed
         # so that we can init a `no_*` complement argument (see below)
         bool_kwargs = {}
-        if origin_type is Literal or (isinstance(field.type, type) and issubclass(field.type, Enum)):
-            if origin_type is Literal:
+        origin_type_is_literal = origin_type is Literal
+        field_typs_is_bool = field.type is bool
```

Comment:
typo

```suggestion
        field_type_is_bool = field.type is bool
```

Reply:
Ah, such a silly issue. Thanks for letting me know